presto-kafka/src/main/java/com/facebook/presto/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,329 @@
+                    "Start mapping '%s' for column '%s' does not equal expected mapping '%s'",
+                    mapping.getStart(),
+                    mapping.getName(),
+                    position));
+            checkArgument(mapping.getEnd() > mapping.getStart(), format(
+                    "End mapping '%s' for column '%s' is less than or equal to start '%s'",
+                    mapping.getEnd(),
+                    mapping.getName(),
+                    mapping.getStart()));
+            position += mapping.getLength();
+        }
+
+        this.buffer = ByteBuffer.allocate(position);
+    }
+
+    private static class ColumnMapping
+    {
+        private final String name;
+        private final Type type;
+        private final FieldType fieldType;
+        private final int start;
+        private final int end;
+
+        public ColumnMapping(EncoderColumnHandle columnHandle)
+        {
+            this.name = columnHandle.getName();
+            this.type = columnHandle.getType();
+
+            this.fieldType = parseFieldType(columnHandle.getDataFormat(), this.name);
+            checkFieldType(this.name, this.type, this.fieldType);
+
+            Optional<String> mapping = Optional.ofNullable(columnHandle.getMapping());
+            if (mapping.isPresent()) {
+                Matcher mappingMatcher = MAPPING_PATTERN.matcher(mapping.get());
+                if (!mappingMatcher.matches()) {
+                    throw new IllegalArgumentException(format("Invalid mapping for column '%s'", this.name));
+                }
+
+                if (mappingMatcher.group(2) != null) {
+                    this.start = parseOffset(mappingMatcher.group(1), "start", this.name);
+                    this.end = parseOffset(mappingMatcher.group(2), "end", this.name);
+                }
+                else {
+                    this.start = parseOffset(mappingMatcher.group(1), "start", this.name);
+                    this.end = this.start + this.fieldType.getSize();
+                }
+            }
+            else {
+                throw new IllegalArgumentException(format("No mapping defined for column '%s'", this.name));
+            }
+        }
+
+        private static int parseOffset(String group, String offsetName, String columnName)
+        {
+            try {
+                return parseInt(group);
+            }
+            catch (NumberFormatException e) {
+                throw new IllegalArgumentException(format("Unable to parse '%s' offset for column '%s'", offsetName, columnName), e);
+            }
+        }
+
+        private static FieldType parseFieldType(String dataFormat, String columnName)
+        {
+            try {
+                if (!dataFormat.equals("")) {
+                    return FieldType.valueOf(dataFormat.toUpperCase(Locale.ENGLISH));
+                }
+                return FieldType.BYTE;
+            }
+            catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)